### PR TITLE
Clang-format review

### PR DESCRIPTION
**Pull request by oscarbailey-xmos**
I ran clang-format recursively on lib_dsp to check the formatting of the current XMOS .clang-format style.

If you see any errors/problems in formatting please comment.

The .clang-format file used is the one in [this pull request](https://github.com/xmos/lib_mp3/pull/72/files) except I changed BreakBeforeBinaryOperators from None to NonAssignment e.g.
this:
```
bool value = aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa +
                     aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa ==
                 aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa;
```
is now
```
bool value = aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
                     + aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
                 == aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa;
```

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  AfterStruct:     false
+  AfterUnion:      false
+  AfterExternBlock: false
+  BeforeCatch:     false
+  BeforeElse:      false
+  IndentBraces:    false
+  SplitEmptyFunction: true
+  SplitEmptyRecord: true
+  SplitEmptyNamespace: true
+BreakBeforeBinaryOperators: NonAssignment
```
Comment (oscarbailey-xmos) [inline .clang-format]: Changed from None

File: AN00209_xCORE-200_DSP_Library/app_complex_fir/src/app_complex_filters.xc [hunk 2]
```diff
@@ -8,36 +8,38 @@
...
-    if (x.re == sum.re && x.im == sum.im) {
-        printf("PASS\n");
-    } else {
-        printf("FAIL\n");
-    }
-            
-    return 0;
+  timer tmr;
+  int   t0, t1;
+tmr:
```
Comment (samchesney) [inline AN00209_xCORE-200_DSP_Library/app_complex_fir/src/app_complex_filters.xc]: It looks like it is currently treating timer reads as labels here.

File: AN00209_xCORE-200_DSP_Library/app_fft/src/app_fft.xc [hunk 3]
```diff
@@ -22,186 +22,222 @@
...
-#define INPUT_FREQ N_FFT_POINTS/8
+#define INPUT_FREQ N_FFT_POINTS / 8
 
 #ifndef INT16_BUFFERS
 #define INT16_BUFFERS 0 // disabled by default. int32_t buffers is default
 #endif
 
 #ifndef TWOREALS
-#define TWOREALS 0 // Processing two real signals with a single complex FFT. Disabled by default
+#define TWOREALS \
```
Comment (samchesney) [inline AN00209_xCORE-200_DSP_Library/app_fft/src/app_fft.xc]: We might want to move this comment by hand as clang-format probably can't do better than this on its own.

File: AN00209_xCORE-200_DSP_Library/app_fft/src/app_fft.xc [hunk 4]
```diff
@@ -22,186 +22,222 @@
...
 #else
-dsp_complex_t  data[N_FFT_POINTS];
+dsp_complex_t data[N_FFT_POINTS];
 #endif
 
 
 /**
- * Experiments with functions that generate sine and cosine signals with a defined number of points
+ * Experiments with functions that generate sine and cosine signals with a
+ *defined number of points
```
Comment (samchesney) [inline AN00209_xCORE-200_DSP_Library/app_fft/src/app_fft.xc]: Is it possible to have it add whitespace after the `*` when it breaks multiline comments like this?

File: AN00209_xCORE-200_DSP_Library/app_complex_fir/src/app_complex_filters.xc (hunk 2 above)
Comment (jh14778) [inline AN00209_xCORE-200_DSP_Library/app_complex_fir/src/app_complex_filters.xc]: Unfortunately you seem to have used a version of clang-format that doesn't support XC.
For this reason it has interpreted `tmr:>t0;` as being a label no a read from a timer.

File: AN00209_xCORE-200_DSP_Library/app_fft_double_buf/src/app_fft_double_buf.xc [hunk 5]
```diff
@@ -16,13 +17,14 @@ mechanism accessing a shared memory area.
 
 The task produce_samples fills one buffer with a stereo signal
 whilst the do_fft task processes the other buffer
-"in place" which means the buffer is also used to store the real part of the result
-Tasks access these buffers via *movable* pointers. These pointers can
-be safely transferred between tasks without any race conditions
-between the tasks using them.
+"in place" which means the buffer is also used to store the real part of the
+result Tasks access these buffers via *movable* pointers. These pointers can be
```
Comment (samchesney) [inline AN00209_xCORE-200_DSP_Library/app_fft_double_buf/src/app_fft_double_buf.xc]: There was a missing full stop after result, which was less important with the original line break arrangement.

File: AN00209_xCORE-200_DSP_Library/app_fft_double_buf/src/app_fft_double_buf.xc [hunk 6]
```diff
@@ -61,11 +63,13 @@ swap function of the interface to synchronise with the do_fft task and swap poin
 #endif
 
 #ifndef TWOREALS
-#define TWOREALS 0 // Processing two real signals with a single complex FFT. Disabled by default
+#define TWOREALS \
```
Comment (samchesney) [inline AN00209_xCORE-200_DSP_Library/app_fft_double_buf/src/app_fft_double_buf.xc]: See above.

File: AN00209_xCORE-200_DSP_Library/app_fft_double_buf/src/app_fft_double_buf.xc [hunk 7]
```diff
@@ -82,47 +86,53 @@ swap function of the interface to synchronise with the do_fft task and swap poin
...
 #endif
 
 /** Union to store blocks of samples for multiple digital signals
-*/
+ */
 typedef union {
-    SIGNAL_ARRAY_TYPE data[NUM_SIGNAL_ARRAYS][N_FFT_POINTS];  // time domain or frequency domain signals
-    SIGNAL_ARRAY_TYPE half_spectra[NUM_SIGNAL_ARRAYS*2][N_FFT_POINTS/2];  // frequency domain half spectra
-} multichannel_sample_block_s ;
+  SIGNAL_ARRAY_TYPE
```
Comment (samchesney) [inline AN00209_xCORE-200_DSP_Library/app_fft_double_buf/src/app_fft_double_buf.xc]: The layout of this union seems a bit odd...

File: AN00209_xCORE-200_DSP_Library/app_adaptive/src/app_adaptive.xc [hunk 8]
```diff
@@ -69,38 +84,48 @@ int32_t fir_state[FIR_FILTER_LENGTH];
...
-      print31( x ); print31( err ); printf( "\n" );
+  // Apply Normalized LMS Filter
+  for (c = 5; c <= 5; c *= 2) {
+    printf("\nNormalized LMS %u\n", c);
+    for (int32_t i = 0; i < FIR_FILTER_LENGTH; ++i)
+      nlms_coeffs[i] = fir_coeffs[i];
+    for (int32_t i = 0; i < FIR_FILTER_LENGTH; ++i)
+      fir_state[i] = 0;
+    for (int32_t i = 0; i < c + 30; ++i) {
+      x = dsp_adaptive_nlms(Q31(0.08),
```
Comment (samchesney) [inline AN00209_xCORE-200_DSP_Library/app_adaptive/src/app_adaptive.xc]: I think we'll want to look at the options to control how it breaks up parameter lists like this.

**Comment by jh14778** [comment]
I think I would like `BinPackParameters` set to false.

This avoids the case of having one parameter per line on long function calls.

This depends on what other people think...

File: AN00209_xCORE-200_DSP_Library/app_fft/src/app_fft.xc (hunk 4 above)
Comment (oscarbailey-xmos) [reply AN00209_xCORE-200_DSP_Library/app_fft/src/app_fft.xc]: This should be the default, not sure what happened here. It works correctly here (https://github.com/xmos/lib_dsp/pull/119/files#diff-9752432e546735d3fb65c7ec27346416R215) for example.

**Pull request closed without merging**